src/utils/change-case.ts

```diff
@@ -0,0 +1,211 @@
+
+/**
+ * Options used for converting strings to pascal/camel case.
+ */
+interface PascalCaseOptions extends Options {
+    mergeAmbiguousCharacters?: boolean;
+}
+
+/**
+ * Options used for converting strings to any case.
+ */
+interface Options {
+    locale?: Locale;
+    split?: (value: string) => string[];
+    /** @deprecated Pass `split: splitSeparateNumbers` instead. */
+    separateNumbers?: boolean;
+    delimiter?: string;
+    prefixCharacters?: string;
+    suffixCharacters?: string;
+}
+
+/**
+ * Split any cased input strings into an array of words.
+ */
+function split(value: string) {
+    let result = value.trim();
+
+    result = result
+        .replace(SPLIT_LOWER_UPPER_RE, SPLIT_REPLACE_VALUE)
+        .replace(SPLIT_UPPER_UPPER_RE, SPLIT_REPLACE_VALUE);
+
+    result = result.replace(DEFAULT_STRIP_REGEXP, "\0");
+
+    let start = 0;
+    let end = result.length;
+
+    // Trim the delimiter from around the output string.
+    while (result.charAt(start) === "\0") start++;
+    if (start === end) return [];
+    while (result.charAt(end - 1) === "\0") end--;
+
+    return result.slice(start, end).split(/\0/g);
+}
+
+/**
+ * Split the input string into an array of words, separating numbers.
+ */
+function splitSeparateNumbers(value: string) {
+    const words = split(value);
+    for (let i = 0; i < words.length; i++) {
+        const word = words[i];
+        if (typeof word === "undefined") continue;
+        const match = SPLIT_SEPARATE_NUMBER_RE.exec(word);
+        if (match) {
+            const elem = match.at(1) ?? match[2];
+            if (typeof elem !== "undefined") {
+                const offset = match.index + elem.length;
+                words.splice(i, 1, word.slice(0, offset), word.slice(offset));
+            }
+        }
+    }
+    return words;
+}
+
+/**
+ * Convert a string to space separated lower case (`foo bar`).
+ */
+function noCase(input: string, options?: Options) {
+    const [prefix, words, suffix] = splitPrefixSuffix(input, options);
+    return (
+        prefix +
+        words
+            .map(lowerFactory(options?.locale))
+            .join(options?.delimiter ?? " ") +
+        suffix
+    );
+}
+
+/**
+ * Convert a string to pascal case (`FooBar`).
+ */
+export function pascalCase(input: string, options?: PascalCaseOptions) {
+    const [prefix, words, suffix] = splitPrefixSuffix(input, options);
+    const lower = lowerFactory(options?.locale);
+    const upper = upperFactory(options?.locale);
+    const transform = options?.mergeAmbiguousCharacters
+        ? capitalCaseTransformFactory(lower, upper)
+        : pascalCaseTransformFactory(lower, upper);
+    return (
+        prefix + words.map(transform).join(options?.delimiter ?? "") + suffix
+    );
+}
+
+/**
+ * Convert a string to snake case (`foo_bar`).
+ */
+export function snakeCase(input: string, options?: Options) {
+    return noCase(input, { delimiter: "_", ...options });
+}
```

Comment:
Let's keep only `pascalCase` and `snakeCase` and their deps

Reply:
redone